CDP4WebServices.API/Services/Supplemental/DomainFileStoreService.cs

```diff
@@ -0,0 +1,226 @@
+// --------------------------------------------------------------------------------------------------------------------
+// <copyright file="DomainFileStoreService.cs" company="RHEA System S.A.">
+//    Copyright (c) 2015-2020 RHEA System S.A.
+//
+//    Author: Sam Gerené, Merlin Bieze, Alex Vorobiev, Naron Phou, Alexander van Delft.
+//
+//    This file is part of CDP4 Web Services Community Edition. 
+//    The CDP4 Web Services Community Edition is the RHEA implementation of ECSS-E-TM-10-25 Annex A and Annex C.
+//
+//    The CDP4 Web Services Community Edition is free software; you can redistribute it and/or
+//    modify it under the terms of the GNU Affero General Public
+//    License as published by the Free Software Foundation; either
+//    version 3 of the License, or (at your option) any later version.
+//
+//    The CDP4 Web Services Community Edition is distributed in the hope that it will be useful,
+//    but WITHOUT ANY WARRANTY; without even the implied warranty of
+//    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+//    Lesser General Public License for more details.
+//
+//    You should have received a copy of the GNU Affero General Public License
+//    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+// </copyright>
+// --------------------------------------------------------------------------------------------------------------------
+
+namespace CDP4WebServices.API.Services
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Data;
+    using System.Linq;
+    using System.Security;
+
+    using CDP4Common.DTO;
+    using CDP4Common.Exceptions;
+
+    using CDP4WebServices.API.Services.Authorization;
+
+    using Npgsql;
+
+    /// <summary>
+    /// The handcoded part of the <see cref="DomainFileStore"/> Service which uses the ORM layer to interact with the data model.
+    /// </summary>
+    public sealed partial class DomainFileStoreService
+    {
+        /// <summary>
+        /// Gets or sets the <see cref="IIterationService"/>.
+        /// </summary>
+        public IIterationService IterationService { get; set; }
+
+        /// <summary>
+        /// 
+        /// </summary>
+        private readonly Dictionary<Type, Func<Guid, Func<DomainFileStore, bool>>> domainFileStoreSelectors = new Dictionary<Type, Func<Guid, Func<DomainFileStore, bool>>>
+        {
+            {
+                typeof(File),
+                guid => 
```

Comment:
Interesting notation, I dont think I have seen this before.what does it mean exactly?

Reply:
It's a Dictionary with a key of 'Type' and its value is a Func<> that returns a Func<>
Its usage was not yet implemented correctly though.
Fixed that, and I'll add a test for it.